libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/NavigationViewModel.kt

```diff
@@ -0,0 +1,23 @@
+package com.mapbox.navigation.dropin
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+
+class NavigationViewModel(
+    accessToken: String,
+    application: Application
+) : AndroidViewModel(application) {
+
+    internal val mapboxNavigation: MapboxNavigation = if (MapboxNavigationProvider.isCreated()) {
+        MapboxNavigationProvider.retrieve()
+    } else {
```

Comment:
Since we plan to control the instance, we should not rely on it being already created. Drop-In UI should reset state of previous instances and spin up it's own. Since it will live in a view model, it will not be recreated on configuration changes which is enough for us.